### DISCUSSION TITLE
Format style "Visual Studio" definition

### DISCUSSION THREAD

**Discussion by JOSE-Sendance** [microsoft/vscode-cpptools#12219] [Q&A] [answered]
I was using the Visual Studio formatting style in VSCode with the cpptools extension.

<img width="933" alt="image" src="https://github.com/microsoft/vscode-cpptools/assets/114486666/62087935-6195-45fd-93eb-22a461732d31">

Now most of my code is formatted in this style. Now I want to run format checks with clang-format for CI/CD but the clang-format executeable does not know the "Visual Studio" style. I tried with the "Microsoft" style but it seems to be a little bit different.

I found this repo online, but it does not exactly have the same style (finds format errors in files formatted with Visual Studio style):
https://gist.github.com/khoa-io/d46aea62294739ada2e54f3a7075a08e

**Comment by bobbrow** [comment, marked as answer]
"Visual Studio" is not an official style that you can supply for the "BasedOnStyle" property in a .clang-format.  The description of how we emulate the "Visual Studio" style is in the setting description that you screengrabbed.

```
BasedOnStyle: LLVM
UseTab: Never
IndentWidth: 4
TabWidth: 4
BreakBeforeBraces: Allman
AllowShortIfStatementsOnASingleLine: false
IndentCaseLabels: false
ColumnLimit: 0
AccessModifierOffset: -4
NamespaceIndentation: All
FixNamespaceComments: false
```
Official documentation here: https://clang.llvm.org/docs/ClangFormatStyleOptions.html

**Reply by bobbrow** [reply]
Do note that we modify `UseTab`, `IndentWidth`, `TabWidth`, and `AccessModifierOffset` values to match your VS Code editor settings. The ones in our setting documentation are defaults based on VS Code's defaults.

**Comment by JOSE-Sendance** [comment]
Yes thanks that worked!

I am using a .clang-format file now, I had to remove the commas, otherwise it does not work.

**Reply by bobbrow** [reply]
Oh, right. 😄 I copied from the setting's description and forgot to remove them.
